server-side/controllers/user.controller.js

```diff
@@ -87,3 +50,262 @@ export const updatedUser = async (req, res,next) => {
+      return next(new Error('Server Error'));
+    }
+  };
+  export const getByEmail = async (req, res) =>
+    {
+      try{
+        const em= req.params.email
+        const user = await User.findOne({ email: em })
+       if(user)
+        res.status(200).send(user)
+      // else
+      //   res.status(404).send('Error email not found');
+      }
+      catch(error){
+      console.error(error);
+      res.status(404).send('Error email not found');
+      // res.status(500).send('Enacorrect to db');
+     }
+    }
+     export const getByEmail1 = async (email) => {
+          try {
+              const user = await User.findOne({ email });
+              return user; // מחזיר את המשתמש אם נמצא, אחרת מחזיר null
+          } catch (error) {
+              console.error('Error: Unable to query database', error);
+              throw new Error('Unable to query database');
+          }
+      };
+  export const getKode = async (req, res)=>{
+    // const us= "st3196420@gmail.com"
+    const us=req.params.email
+    try{
+      const user = await getByEmail1( us);
+      if(user){
+        // sendTemporaryPasswordToCustomerEmail(user)
+        // sendEmail(user.email)//, req.params.password
+        sendEmail(user)
+        res.status(200).send( "We will send you a verification code to " + user.email+" email" )
+      }
+      else
+      {
+        res.status(404).send("Error email not found")
+      }
+    }
+    catch{
+      res.status(500).send('Internal Server Error');
+
+    }
+  
+  
+  }
+  export const resetPassword=async (req,res,next)=>{
+    const { email,password } = req.body;
+  
+    if (!email||!password) {
+      return res.status(404).send('email and password are required');
+    }
+    try {
+      const user = await User.findOne({ email });
+  
+      if (!user) {
+        return res.status(404).send('User not found');
+      }
+  
+      const hashedPassword = await bcrypt.hash(password, 10);
+  
+      user.password = hashedPassword;
+      await user.save();
+  
+      res.status(200).send('Password updated successfully');
+    } catch (error) {
+      res.status(500).send('Server error');
+    }
+  
+  }
+
+
+
+
+
+
+export const addUser = async (req, res) => {
+  const { name, password, email, } = req.body;//googleId
+  if(password){
+    try {
+      const hashedPassword = await bcrypt.hash(password, 10);
+      const newUser = new User({
+        name,
+        password: hashedPassword,
+        email,
+      });
+      await newUser.save();
+      res.send('Data saved successfully!');
+    } catch (err) {
+      console.error(err);
+      res.status(500).send('Error saving user');
+    }
+
+  }
+  // else if(googleId){
+
+  // }
+
+};
+
+export const deleteUser = async (req, res) => {
+  try {
+    const idParams = req.params.id;
+    const user = await User.findByIdAndDelete(idParams);
+    if (!user) {
+      res.status(404).send('User not found');
+      return;
+    }
+    res.send('User deleted successfully!');
```

Comment:
add status code & move text to constants file

Reply:
it is not my function

Reply:
it is not my function

Reply:
so why do you have this changes in your branch?